File: ingestWorker/internal/models/submission.go
package models

import "time"

type SubmissionStatus string

const (
	SubmissionStatusUploaded SubmissionStatus = "uploaded"
	SubmissionStatusParsing  SubmissionStatus = "parsing"
	SubmissionStatusParsed   SubmissionStatus = "parsed"
	SubmissionStatusFailed   SubmissionStatus = "failed"
)

type Submission struct {
	ID             string
	LabID          string
	StudentID      string
	Status         SubmissionStatus
	SourceFileName string
	MimeType       string
	StorageKey     string
	SubmittedAt    time.Time
}

File: ApiService/Domain/Submissions/Submission.cs
namespace Domain.Submissions;

public sealed record Submission(

    Guid SubmissionId,

    Guid LabId,

    Guid StudentId,

    SubmissionStatus Status,

    string MimeType,

    string SourceFileName,

    string StorageKey,
    
    DateTimeOffset SubmittedAt
);   